From 7d2f6b6e31fad2c96c6fc55c17e79f8c96b4bfcf Mon Sep 17 00:00:00 2001
From: Yeoreum Yun <yeoreum.yun@arm.com>
Date: Mon, 10 Aug 2026 15:40:06 +0100
Subject: [PATCH] DynamicTablesPkg: fix SmbiosType2Genertor build failure with
 clang

BaseboardCount type is UINT32. Using MAX_UINTN in the comparision
triggers below build error:

  type 'UINT32' (aka 'unsigned int') is always false
    [-Werror,-Wtautological-constant-out-of-range-compare]
  199 |       (BaseboardCount > (MAX_UINTN / sizeof (*CmObjectList))))

To address this, use MAX_UINT32 instead of MAX_UINTN.

Signed-off-by: Yeoreum Yun <yeoreum.yun@arm.com>

Upstream-Status: Backport [7d2f6b6e31fad2c96c6fc55c17e79f8c96b4bfcf]
Signed-off-by: Jon Mason <jon.mason@arm.com>
---
 .../Library/Smbios/SmbiosType2Lib/SmbiosType2Generator.c      | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/DynamicTablesPkg/Library/Smbios/SmbiosType2Lib/SmbiosType2Generator.c b/DynamicTablesPkg/Library/Smbios/SmbiosType2Lib/SmbiosType2Generator.c
index 1250ef8059e9..5345d2b6d233 100644
--- a/DynamicTablesPkg/Library/Smbios/SmbiosType2Lib/SmbiosType2Generator.c
+++ b/DynamicTablesPkg/Library/Smbios/SmbiosType2Lib/SmbiosType2Generator.c
@@ -195,8 +195,8 @@ BuildSmbiosType2TableEx (
   }
 
   if ((BaseboardInfo == NULL) ||
-      (BaseboardCount > (MAX_UINTN / sizeof (*TableList))) ||
-      (BaseboardCount > (MAX_UINTN / sizeof (*CmObjectList))))
+      (BaseboardCount > (MAX_UINT32 / sizeof (*TableList))) ||
+      (BaseboardCount > (MAX_UINT32 / sizeof (*CmObjectList))))
   {
     DEBUG ((DEBUG_ERROR, "%a: Invalid Baseboard object list\n", __func__));
     return EFI_INVALID_PARAMETER;
-- 
2.39.5

